docs(proxy) add docs for X-Forwarded-Prefix (#2137)
Kong/docs.konghq.com: app/2.1.x/proxy.md

  **trusted** addresses, the request header with the same name gets forwarded
  if provided. Otherwise, the value of the `$server_port` variable provided by
  [ngx_http_core_module][ngx-server-port-variable] will be used.
- `X-Forwarded-Prefix: <path>`, where `<path>` is the path of the request which
  was accepted by Kong. In the case where `$realip_remote_addr` is one of the
  **trusted** addresses, the request header with the same name gets forwarded
  if provided. Otherwise, the value of the `$request_uri` variable (with

>   query string stripped) provided by [ngx_http_core_module][ngx-server-port-variable]

Reviewer: ```suggestion
  the query string stripped) provided by [ngx_http_core_module][ngx-server-port-variable]
```

>   will be used. **Note**: we do not currently do any normalization on the request
>   path except return `"/"` on empty path.

Reviewer: ```suggestion
  will be used. **Note**: Kong will return `"/"` for an empty path, but it does not do any other normalization on the request path.
```
